class.jetpack.php

```diff
@@ -5318,16 +5318,23 @@ function verify_xml_rpc_signature() {
 		}
 
 		// Let's see if this is onboarding. In such case, use user token type and the provided user id.
-		if ( isset( $this->HTTP_RAW_POST_DATA ) ) {
-			$jpo = json_decode( $this->HTTP_RAW_POST_DATA );
+		if ( isset( $this->HTTP_RAW_POST_DATA ) || ! empty( $_GET['onboarding'] ) ) {
+			if ( ! empty( $_GET['onboarding'] ) ) {
+				$jpo = $_GET;
+			} else {
+				$jpo = json_decode( $this->HTTP_RAW_POST_DATA, true );
+			}
```

Comment:
What about setting `$jpo` to `$_GET['onboarding']` or `json_decode( $this->HTTP_RAW_POST_DATA->onboarding, true )`, respectively?

Reply:
Didn't want to access properties of `$this->HTTP_RAW_POST_DATA` directly. I think I'm good keeping it as-is.